src/Polly.Core/Simmy/Fault/FaultChaosStrategy.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Simmy.Fault;
+internal class FaultChaosStrategy<T> : MonkeyStrategy<T>
+{
+    private readonly ResilienceStrategyTelemetry _telemetry;
+
+    public FaultChaosStrategy(FaultStrategyOptions options, ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        if (options.Fault is null && options.FaultGenerator is null)
+        {
+            throw new InvalidOperationException("Either Fault or FaultGenerator is required.");
+        }
+
+        _telemetry = telemetry;
+        Fault = options.Fault;
+        OnFaultInjected = options.OnFaultInjected;
+        FaultGenerator = options.FaultGenerator is not null ? options.FaultGenerator : (_) => new(options.Fault);
+    }
+
+    public Func<OnFaultInjectedArguments, ValueTask>? OnFaultInjected { get; }
+
+    public Func<FaultGeneratorArguments, ValueTask<Exception?>>? FaultGenerator { get; }
```

Comment:
This doesn't appear to need to be nullable as the constructor makes it non-null. That then means you can delete the `!` on line 34.